elements/lisk-dpos/src/delegates_list.ts

```diff
@@ -121,26 +171,166 @@ export const getForgerPublicKeysForRound = async (
+		height: number,
+		stateStore: StateStore,
+		roundOffset: number = DEFAULT_ROUND_OFFSET,
+	): Promise<void> {
+		const round = this.rounds.calcRound(height) + roundOffset;
+		debug(`Creating vote weight snapshot for round: ${round}`);
+		// This list is before executing the current block in process
+		const originalDelegates = await this.chain.dataAccess.getDelegates();
+
+		// Merge updated delegate accounts
+		const updatedAccounts = stateStore.account.getUpdated();
+		// tslint:disable-next-line readonly-keyword
+		const updatedAccountsMap: { [address: string]: Account } = {};
+		// Convert updated accounts to map for better search
+		for (const account of updatedAccounts) {
+			// Insert only if account is a delegate
+			if (account.username) {
+				updatedAccountsMap[account.address] = account;
+			}
+		}
+		// Inject delegate account if it doesn't exist
+		for (const delegate of originalDelegates) {
+			if (updatedAccountsMap[delegate.address] === undefined) {
+				updatedAccountsMap[delegate.address] = delegate;
+			}
+		}
+		const delegates = [...Object.values(updatedAccountsMap)];
+
+		// Update totalVotesReceived to voteWeight equivalent before sorting
+		for (const account of delegates) {
+			// If the account is being punished, then consider them as vote weight 0
+			if (isCurrentlyPunished(height, account.delegate.pomHeights)) {
+				account.totalVotesReceived = BigInt(0);
+				continue;
+			}
+			const selfVote = account.votes.find(
+				vote => vote.delegateAddress === account.address,
+			);
+			const cappedValue =
+				(selfVote?.amount ?? BigInt(0)) * BigInt(this.voteWeightCapRate);
+			if (account.totalVotesReceived > cappedValue) {
+				account.totalVotesReceived = cappedValue;
+			}
+		}
+
+		delegates.sort((a, b) => {
+			const diff = b.totalVotesReceived - a.totalVotesReceived;
+			if (diff > BigInt(0)) {
+				return 1;
+			}
+			if (diff < BigInt(0)) {
+				return -1;
+			}
+
+			return a.address.localeCompare(b.address, 'en');
+		});
+
+		const activeDelegates = [];
+		const standbyDelegates = [];
+		for (const account of delegates) {
+			// If the account is banned, do not include in the list
+			if (account.delegate.isBanned) {
+				continue;
+			}
+
+			// Select active delegate first
+			if (activeDelegates.length < this.activeDelegates) {
+				activeDelegates.push({
+					address: account.address,
+					voteWeight: account.totalVotesReceived.toString(),
+				});
+				continue;
+			}
+
+			// If account has more than threshold, save it as standby
+			if (account.totalVotesReceived >= this.standbyThreshold) {
+				standbyDelegates.push({
+					address: account.address,
+					voteWeight: account.totalVotesReceived.toString(),
+				});
+				continue;
+			}
+
+			// From here, it's below threshold
+			// Below threshold, but prepared array does not have enough slected delegate
+			if (
+				activeDelegates.length + standbyDelegates.length <
+				this.activeDelegates + this.standbyDelegates
```

Comment:
```suggestion
				standbyDelegates.length < this.standbyDelegates
```

Reply:
I think that if we get here in the loop, then `activeDelegates.length = this.activeDelegates`. no?